converter/src/main/java/gov/cms/qpp/conversion/validate/CpcMeasureDataValidator.java

```diff
@@ -0,0 +1,149 @@
+package gov.cms.qpp.conversion.validate;
+
+import gov.cms.qpp.conversion.decode.MeasureDataDecoder;
+import gov.cms.qpp.conversion.model.Node;
+import gov.cms.qpp.conversion.model.Program;
+import gov.cms.qpp.conversion.model.TemplateId;
+import gov.cms.qpp.conversion.model.Validator;
+import gov.cms.qpp.conversion.model.error.Detail;
+import gov.cms.qpp.conversion.model.error.ErrorCode;
+import gov.cms.qpp.conversion.model.error.LocalizedError;
+import gov.cms.qpp.conversion.model.validation.MeasureConfig;
+import gov.cms.qpp.conversion.model.validation.MeasureConfigs;
+import gov.cms.qpp.conversion.model.validation.SupplementalData;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+/**
+ * Validates a Sub Population's Measure Data for the CPC Plus program entity
+ */
+@Validator(value = TemplateId.MEASURE_DATA_CMS_V2, program = Program.CPC)
+public class CpcMeasureDataValidator extends NodeValidator {
+
+	/**
+	 * Validates a single measure data sub population
+	 *
+	 * @param node The node to validate.
+	 */
+	@Override
+	protected void internalValidateSingleNode(Node node) {
+		validateSupplementalDataByType(node);
+	}
+
+	/**
+	 * Categorizes each validation by specified Supplemental Types
+	 *
+	 * @param node current measure data node
+	 */
+	private void validateSupplementalDataByType(Node node) {
+		Map<String, TemplateId> nodeTypeToTemplateIdMap = SupplementalData.getSupplementalTypeMapToTemplateId();
+		for (String supplementalDataType: nodeTypeToTemplateIdMap.keySet()) {
+			validateAllSupplementalNodesOfSpecifiedType(
+					node, nodeTypeToTemplateIdMap.get(supplementalDataType), supplementalDataType);
+		}
+	}
+
+	/**
+	 * Validates all of the nodes under one specified Supplemental Type
+	 *
+	 * @param node Parent node of the nodes to be validated
+	 * @param currSupplementalDataTemplateId Template Id of the nodes to be validated
+	 * @param supplementalDataType current data type to be validated
+	 */
+	private void validateAllSupplementalNodesOfSpecifiedType(Node node,
+															 TemplateId currSupplementalDataTemplateId,
+															 String supplementalDataType) {
+		List<Node> supplementalDataNodes =
```

Comment:
Consider using a `Set` since all the nodes should be unique.

Reply:
Will look into this.